.github/autolabeler.yml

```diff
@@ -0,0 +1,83 @@
+
+# The following paragraph was taken from the Apache Spark project.
+#   see: https://github.com/apache/spark/blob/master/.github/autolabeler.yml
+# Bot page: https://github.com/apps/probot-autolabeler
+# The matching patterns follow the .gitignore spec.
+# See: https://git-scm.com/docs/gitignore#_pattern_format
+# Also, note that the plugin uses 'ignore' package. See also
+# https://github.com/kaelzhang/node-ignore
+
+INFRA:
+  - ".baseline/"
+  - ".github/"
+  - "/dev/"
+  - ".asf.yaml"
+  - ".gitattributes"
+  - ".gitignore"
+  - "baseline.gradle"
+  - "deploy.gradle"
+  - "jitpack.yml"
+  - "site/mkdocs.yml"
+  - "travis.yml"
+BUILD:
+  - "/dev/"
+  - "/project/"
+  - "/assembly/"
+  - "*pom.xml"
+  - "/bin/docker-image-tool.sh"
+  - "/bin/find-spark-home*"
+  - "gradle/"
+  - "**.gradle"
+BASH:
+  - "dev/check-license"
+  - "**.sh"
+DOCS:
+  - "site/"
+  - "!/site/mkdocs.yml"
+  - "**CHANGELOG.md"
+  - "**LICENSE"
+  - "**NOTICE"
+  - "**README.md"
+EXAMPLES:
+  - "examples/"
+COMMON:
+  - "common/"
+API:
+  - "api/"
+  - "python/iceberg/api/"
+  - "python/tests/api/"
+CORE:
+  - "**/core/"
```

Comment:
What does this match?

Reply:
This matches all files underneath a directory called core, that might be nested underneath 0 or more directories. I can remove them if we think that complicates things.

Reply:
I don't think we want to match subdirectories, just the top-level ones.

Reply:
Ok. That sounds good to me. I was thinking in the case where `python` might get expanded out more in the future, but I am now leaning more towards your simpler is better approach.

I'll remove all of the matches for subdirectories.